### PR TITLE
Add validation for #1592; also add a test for UUID case (failing)

### LINKED ISSUE #1592: More gRPC `ValueCodec` implementations should verify validity of data decoded (ByteBuffer->proto Value)
(note: inspired by #1577 )

Looks like following `ValueCodec`s only enforce size limitations on encoding, but not decoding:

* `DecimalCodec` (actually has some checking, might be ok)
* `InetCodec`
* `VarintCodec` (also has some checking)

(validation was added to `UuidCodec` and is helping pinpoint apparent conversion issue/corruption with UUID values)

It seems useful to add such checks as they are very low cost and can prevent corrupt data from being served downstream: this becomes particularly important with StargateV2 when gRPC bridge will becomes the universal interface.

*EDIT*: Actually codecs for `Decimal` and `VarInt` have some checking, and seem to coerce 0-length data into `null`.
There is no comment explaining this behavior: looks like bit of defensive programming. But what is not clear is whether this is something that SHOULD or SHOULD NOT be done -- it could help or it might mask an actual problem?

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**Which issue(s) this PR fixes**:

Fixes #1592 (add more validation) but also related to #1577

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: grpc/src/main/java/io/stargate/grpc/codec/DecimalCodec.java [hunk 1]
```diff
@@ -43,9 +43,7 @@ public ByteBuffer encode(@NonNull Value value, @NonNull ColumnType type) {
 
   @Override
   public Value decode(@NonNull ByteBuffer bytes, @NonNull ColumnType type) {
-    if (bytes.remaining() == 0) {
-      return null;
```
Comment (tatu-at-datastax) [inline grpc/src/main/java/io/stargate/grpc/codec/DecimalCodec.java:47]: This could not have worked: protobuf/grpc code will throw NPE upon getting `null`. Null handling is done by caller, before this method gets called.

File: grpc/src/main/java/io/stargate/grpc/codec/VarintCodec.java [hunk 2]
```diff
@@ -36,7 +36,8 @@ public ByteBuffer encode(@NonNull Value value, @NonNull ColumnType type) {
   @Override
   public Value decode(@NonNull ByteBuffer bytes, @NonNull ColumnType type) {
     if (bytes.remaining() == 0) {
-      return null;
```
Comment (tatu-at-datastax) [inline grpc/src/main/java/io/stargate/grpc/codec/VarintCodec.java:39]: This could not have worked: protobuf/grpc code will throw NPE upon getting `null`. Null handling is done by caller, before this method gets called.

**Comment by tatu-at-datastax** [comment]
@EricBorczuk @mpenick Ready for review, I think -- adds checks in codecs, a new failing test (that is Disabled to pass CI).

May want to add tests for `InetCodec`, `DecimalCodec` and `VarintCodec` as the next step (along with possible fixes as needed).

Would also be good to have this tested at unit test level but I don't yet know things well enough to try that.

File: testing/src/main/java/io/stargate/it/grpc/ExecuteQueryTest.java [hunk 3]
```diff
@@ -184,6 +188,32 @@ public void simpleQueryWithPaging(@TestKeyspace CqlIdentifier keyspace)
     assertThat(rs.getPagingState()).isNotNull();
   }
 
+  @Test
+  @Disabled("Until issue #1577 fixed")
```
Comment (tatu-at-datastax) [inline testing/src/main/java/io/stargate/it/grpc/ExecuteQueryTest.java:192]: Could separate into different PR (one with the fix), as an alternative. But thought this is easier way to share.

**Review by mpenick — APPROVED** [review]

**Review by EricBorczuk — APPROVED** [review]
This all looks good! Asked one question about the test for UUID

File: testing/src/main/java/io/stargate/it/grpc/ExecuteQueryTest.java [hunk 4]
```diff
@@ -184,6 +188,32 @@ public void simpleQueryWithPaging(@TestKeyspace CqlIdentifier keyspace)
...
+  public void simpleQueryWithPagingUuidKey(@TestKeyspace CqlIdentifier keyspace) {
+    StargateBlockingStub stub = stubWithCallCredentials();
+    final String uuid = UUID.randomUUID().toString();
+
+    for (int i = 0; i < 5; i++) {
+      stub.executeQuery(
+          cqlQuery(
+              String.format("INSERT INTO test_uuid (id, value) VALUES (%s, %d)", uuid, i),
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply testing/src/main/java/io/stargate/it/grpc/ExecuteQueryTest.java:200]: Might be a good idea if it's easy to generalize? I do have a PR to fix the actual issue, along with unit test so will think about this one after getting that in.
